src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs

```diff
@@ -1315,11 +1318,139 @@ public async Task Reset_DuringRequestBody_Resets(HttpContext httpContext)
             }
         }
 
+        public Task Goaway(HttpContext httpContext)
+        {
+            httpContext.Response.Headers["Connection"] = "close";
+            return Task.CompletedTask;
+        }
+
+        private TaskCompletionSource _completeAsync = new TaskCompletionSource();
```

Comment:
I see you're living dangerously not using TaskCreationOptions.RunContinuationsAsynchronously.